test/common/test_runner.go

```diff
@@ -86,18 +85,10 @@ func Setup(t *testing.T, runInParallel bool) *Client {
 	if err != nil {
 		t.Fatalf("Couldn't initialize clients: %v", err)
 	}
-
 	CreateNamespaceIfNeeded(t, client, namespace)
 
-	// Disallow manually interrupting the tests.
-	// TODO(Fredy-Z): t.Skip() can only be called on its own goroutine.
-	//                Investigate if there is other way to gracefully terminte the tests in the middle.
-	c := make(chan os.Signal)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
-	go func() {
-		<-c
-		fmt.Printf("Test %q running, please don't interrupt...\n", t.Name())
-	}()
+	// Clean up the resources for this test if it's interrupted.
```

Comment:
While I +1 this, this sounds pretty unrelated and I would prefer having it on a separate PR (specially for tracking purposes).